### PR TITLE
Vm controller unit tests

### PR DESCRIPTION
**Pull request by outscale-vbr**
* Handle link and unlink volume
* Handle add vm as backend
* Handle attach eip
* Handle add sg with vm sgmember
* Update volume unit test
* Add vm unit-test

### CODE REVIEW

**Comment by outscale-vbr** [comment]
Unit-test passed:
https://github.com/outscale-vbr/cluster-api-provider-outscale/runs/7744334384?check_suite_focus=true

File: controllers/oscmachine_vm_controller.go [hunk 1]
```diff
@@ -364,27 +378,30 @@ func reconcileVm(ctx context.Context, clusterScope *scope.ClusterScope, machineS
...
+			machineScope.Info("Get fromPortRange", "fromPortRange", fromPortRange)
 			toPortRange := loadBalancerSpec.Listener.BackendPort
-			machineScope.Info("#### Get ToPortRange ###", "ToPortRange", toPortRange)
-			flow := "Outbound"
+			machineScope.Info("Get ToPortRange", "ToPortRange", toPortRange)
 			loadBalancerSecurityGroupClusterScopeName := loadBalancerSecurityGroupName + "-" + clusterScope.GetUID()
 			associateSecurityGroupId := securityGroupsRef.ResourceMap[loadBalancerSecurityGroupClusterScopeName]
 			machineScope.Info("Get sg", "associateSecurityGroupId", associateSecurityGroupId)
-			_, err = securityGroupSvc.CreateSecurityGroupRule(associateSecurityGroupId, flow, ipProtocol, "", securityGroupIds[0], fromPortRange, toPortRange)
+			_, err = securityGroupSvc.CreateSecurityGroupRule(associateSecurityGroupId, "Outbound", ipProtocol, "", securityGroupIds[0], fromPortRange, toPortRange)
```
[the remark this answers is not in the document]
Comment (outscale-vbr) [reply controllers/oscmachine_vm_controller.go:397]: @outscale-mdr https://docs.outscale.com/fr/userguide/%C3%80-propos-des-security-groups-dans-LBU.html
You have to set inbound and outbound specific securityGroup associate with securityGroupMember for lbu (https://docs.outscale.com/fr/userguide/%C3%80-propos-des-security-groups-dans-LBU.html)

[the remark this answers is not in the document]
Comment (outscale-vbr) [reply controllers/oscmachine_vm_controller.go:397]: Yes, there is a default rules which  allow outbound traffic, but you have to set  outbound security group rule (in lbu sg) for securityGroup (securityGroupMember) associate to your backend instance associate with lbu.
And you have to set inbound  security group rule in each sg backend instance associate with lbu securityGroupMember.

**Review by outscale-mdr — APPROVED** [review]

**Comment by outscale-vbr** [comment]
close https://github.com/outscale-dev/cluster-api-provider-outscale/issues/29

**Pull request merged**